src/test/java/dk/kb/util/yaml/YAMLTest.java

```diff
@@ -89,6 +89,107 @@ public void testDefault() throws IOException {
                      "Requesting a non-existing integer with a default value should work");
     }
     
+    
+    
+    @Test
+    public void testNestedExtrapolated() throws IOException {
+        YAML yaml = YAML.resolveLayeredConfigs("testExtrapolated.yml").extrapolateSystemProperties(true);
+        assertEquals(System.getProperties().getProperty("user.home")+" Hello World", yaml.getString("test.somestring"),
+                     "Nested request for string should be supported");
+    }
+    
+    @Test
+    public void testRootExtrapolated() throws IOException {
+        YAML yaml = YAML.resolveLayeredConfigs("testExtrapolated.yml").getSubMap("test").extrapolateSystemProperties(true);
+        assertEquals(System.getProperties().getProperty("user.home")+" Hello World", yaml.getString("somestring"),
+                     "Direct request for string from resolved root should be supported");
+    }
+    
+    @Test
+    public void testArrayExtrapolated() throws IOException {
+        YAML yaml = YAML.resolveLayeredConfigs("testExtrapolated.yml").getSubMap("test").extrapolateSystemProperties(true);
+        assertEquals("[a, "+System.getProperties().getProperty("user.name")+", c]", yaml.getList("arrayofstrings").toString(),
+                     "Arrays of strings should be supported");
+    }
+    
+    @Test
+    public void testKeptPathExtrapolated() throws IOException {
+        YAML yaml = YAML.resolveLayeredConfigs("testExtrapolated.yml").getSubMap("test").extrapolateSystemProperties(true);
+        
+        assertEquals("{nested.sublevel2string=${user.name}}", yaml.getSubMap("nested",true).toString(),
+                     "When we get map with subkeys preserved, we should see the nested previs. Extrapolation does NOT happen on toString");
```

Comment:
Nice feature. This should work with the `yamlToString`-feature, so that non-expanded properties are preserved in the re-parsed YAML, right?

Reply:
This have been reworked now.